### PR TITLE
chore: add sdkProxy version to events metadata

### PR DESCRIPTION
**Pull request by suthar26**
- add sdkProxy version to the events going out to the events api

### CODE REVIEW

File: http_endpoints.go [hunk 1]
```diff
@@ -75,6 +76,11 @@ func Track() gin.HandlerFunc {
 
 		event := getEventFromBody(c)
 		for _, e := range event.Events {
+			if e.MetaData == nil {
+				e.MetaData = make(map[string]interface{})
+			}
+			e.MetaData["sdkProxy"] = Version
+			fmt.Println("event", e)
```
Comment (kaushalkapasi) [inline http_endpoints.go:83]:
```suggestion
```

Comment (suthar26) [reply http_endpoints.go:83]: fixed, will manually test this once again for custom events

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -66,7 +66,13 @@ jobs:
...
       - uses: actions/setup-go@v5
         with:
-          go-version: '1.20'
+          go-version: "1.20"
+
+      - name: Update version.go file
+        run: |
+          echo "package local_bucketing_proxy" > version.go
+          echo "" >> version.go
+          echo "const Version = \"${{ steps.prepare-release.outputs.next-release-tag }}\"" >> version.go
```
Comment (JamieSinn) [inline .github/workflows/release.yml:75]: just run a `sed` command to replace the version line.

File: http_endpoints.go (hunk 1 above)
Comment (JamieSinn) [inline http_endpoints.go:83]:
```suggestion
```

File: http_endpoints.go [hunk 3]
```diff
@@ -87,23 +93,82 @@ func BatchEvents() gin.HandlerFunc {
 	return func(c *gin.Context) {
 		client := c.Value("devcycle").(*devcycle.Client)
 
+		body, err := io.ReadAll(c.Request.Body)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"message": "Error reading request body: " + err.Error()})
+			return
```
Comment (JamieSinn) [inline http_endpoints.go:99]: This error handling needs to match the actual endpoint on the events api.

File: http_endpoints.go [hunk 4]
```diff
@@ -87,23 +93,82 @@ func BatchEvents() gin.HandlerFunc {
...
+		}
+
+		var batchEvents map[string]interface{}
+		err = json.Unmarshal(body, &batchEvents)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"message": "Error unmarshaling request body: " + err.Error()})
+			return
```
Comment (JamieSinn) [inline http_endpoints.go:106]: More error handling we need to match the actual events API

File: http_endpoints.go (hunk 3 above)
Comment (JamieSinn) [reply http_endpoints.go:99]: I would probably set this as a 500 not 400

File: http_endpoints.go (hunk 4 above)
Comment (JamieSinn) [reply http_endpoints.go:106]: Also a 500; not 400

**Review by JamieSinn — APPROVED** [review]

**Pull request merged**